### PR TITLE
Feature/add dataset

### PR DESCRIPTION
**Pull request by AlexandreMessmer**
Prepare the codebase for the complete dataset. 
Sideway: Replace the HashMaps for Firebase with the serialization.

### CODE REVIEW

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/database/fakes/FakeReviewsRepository.kt [hunk 1]
```diff
@@ -122,38 +124,9 @@ class FakeReviewsRepository @Inject constructor() : ReviewRepository {
         reviewList[0].dislikers = listOf(FAKE_UID_3, FAKE_UID_4)
     }
 
-    override suspend fun take(number: Long): QuerySnapshot =
-        Mockito.mock(QuerySnapshot::class.java)
-
-    override suspend fun getById(id: String): DocumentSnapshot =
-        Mockito.mock(DocumentSnapshot::class.java)
-
-    override fun remove(id: String): Task<Void> {
```
Comment (mbouilloud) [inline app/src/androidTest/java/com/github/sdp/ratemyepfl/database/fakes/FakeReviewsRepository.kt:131]: You should keep this function here, or the tests won't work.

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/database/util/ItemTest.kt [hunk 2]
```diff
@@ -36,27 +37,4 @@ class ItemTest {
         collection = db.collection("itemsTest")
         repository = RepositoryImpl(db, "itemsTest") { it.toItem() }
     }
-
-    @Test
-    fun test() = runTest {
```
Comment (mbouilloud) [inline app/src/androidTest/java/com/github/sdp/ratemyepfl/database/util/ItemTest.kt:41]: This file is empty now.

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/database/fakes/FakeReviewsRepository.kt (hunk 1 above)
Comment (AlexandreMessmer) [reply app/src/androidTest/java/com/github/sdp/ratemyepfl/database/fakes/FakeReviewsRepository.kt:131]: I guess it's one of the methods that is implemented in the FakeRepository. Each fake now inherits it (it is much simpler to maintain).

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/database/util/ItemTest.kt (hunk 2 above)
Comment (AlexandreMessmer) [reply app/src/androidTest/java/com/github/sdp/ratemyepfl/database/util/ItemTest.kt:41]: Yes, good point. I'll remove it.

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/database/fakes/FakeReviewsRepository.kt (hunk 1 above)
Comment (mbouilloud) [reply app/src/androidTest/java/com/github/sdp/ratemyepfl/database/fakes/FakeReviewsRepository.kt:131]: Yes but this function actually does something here. In the FakeRepository you just mock.

Comment (mbouilloud) [reply app/src/androidTest/java/com/github/sdp/ratemyepfl/database/fakes/FakeReviewsRepository.kt:131]: I am talking about the remove function.

**Review by jmettler27 — APPROVED** [review]
We now got rid of a lot of overhead, thanks.

**Review by NicolasMatekalo — APPROVED** [review]

**Pull request merged**